### PR TITLE
macOS CMake check fails

### PR DESCRIPTION
**Pull request by HofiOne**
- CI: Re-enabled cmake check build step in the CI Action
- cmake/autotools: added the missing -flat_namespace linker option to cmake builds and made this option conditional, it is not needed for production build just for the current mocking solution in the criterion tests

Fixes: https://github.com/syslog-ng/syslog-ng/issues/4625

Signed-off-by: hofione@gmail.com

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

File: cmake/add_tests.cmake [hunk 1]
```diff
@@ -19,55 +19,62 @@
...
+  cmake_parse_arguments(ADD_UNIT_TEST "CRITERION;LIBTEST" "TARGET" "SOURCES;DEPENDS;INCLUDES;MOCKS" ${ARGN})
 
-  if (NOT ADD_UNIT_TEST_SOURCES)
+  if(NOT ADD_UNIT_TEST_SOURCES)
     set(ADD_UNIT_TEST_SOURCES "${ADD_UNIT_TEST_TARGET}.c")
   endif()
 
+  if(ADD_UNIT_TEST_MOCKS)
+    set(ADD_UNIT_TEST_MOCKS_LIB "${ADD_UNIT_TEST_TARGET}_mocks")
+    add_library(${ADD_UNIT_TEST_MOCKS_LIB} SHARED "${ADD_UNIT_TEST_MOCKS}")
```
Comment (MrAnno) [inline cmake/add_tests.cmake:44]: Can you explain please, how this works?

When a mock is defined, it looks like both the original and the mocked symbol will exist in 2 different shared libraries. Which one will be used and why?

Comment (HofiOne) [reply cmake/add_tests.cmake:44]: Unlikely in the case of static linking, seems for shared libraries
- the link order decides where a unit-referenced function will be searched for first
so, the key here is 
```
   target_link_libraries(${ADD_UNIT_TEST_MOCKS_LIB} syslog-ng)
  endif()
  ...
  target_link_libraries(${ADD_UNIT_TEST_TARGET} ${ADD_UNIT_TEST_DEPENDS} ${ADD_UNIT_TEST_MOCKS_LIB} syslog-ng)

```
- the mocked functions in the mocking shared library are looking for and using any referenced functions in itself first
that is exactly our intended behavior.

(Telling the truth, I did not play along with all the variants, I was satisfied with the result that worked both on Linux and macOS.)

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

File: cmake/add_tests.cmake (hunk 1 above)
Comment (HofiOne) [reply cmake/add_tests.cmake]: This whole mock separation is revoked currently

**Review by MrAnno — APPROVED** [review]

**Pull request merged**